node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -561,64 +574,87 @@ async fn connect_to_validators<Context>(
 
 /// Advertise collation to the given `peer`.
 ///
-/// This will only advertise a collation if there exists one for the given `relay_parent` and the given `peer` is
-/// set as validator for our para at the given `relay_parent`.
+/// This will only advertise a collation if there exists at least one for the given
+/// `relay_parent` and the given `peer` is set as validator for our para at the given `relay_parent`.
+///
+/// We also make sure not to advertise the same collation multiple times to the same validator.
 #[overseer::contextbounds(CollatorProtocol, prefix = self::overseer)]
 async fn advertise_collation<Context>(
 	ctx: &mut Context,
-	state: &mut State,
 	relay_parent: Hash,
-	peer: PeerId,
+	per_relay_parent: &mut PerRelayParent,
+	peer: &PeerId,
+	protocol_version: CollationVersion,
+	peer_ids: &HashMap<PeerId, HashSet<AuthorityDiscoveryId>>,
+	metrics: &Metrics,
 ) {
-	let should_advertise = state
-		.our_validators_groups
-		.get(&relay_parent)
-		.map(|g| g.should_advertise_to(&state.peer_ids, &peer))
-		.unwrap_or(false);
+	for (candidate_hash, collation) in per_relay_parent.collations.iter_mut() {
+		// Check that peer will be able to request the collation.
+		if let CollationVersion::V1 = protocol_version {
+			if per_relay_parent.prospective_parachains_mode.is_enabled() {
+				gum::debug!(
```

Comment:
Given the control flow leading to `advertise_collation` from `distribute_collation`, we will get a lot of `debug!` logs when prospective parachains are first activated, in the case that there are any v1 peers still connected. I would recommend downgrading this to a `trace!` as we expect to hit this line under normal conditions.